platforms/chibios/eeprom_stm32_defs.h

```diff
@@ -51,7 +51,7 @@
 #        define FEE_MCU_FLASH_SIZE 128 // Size in Kb
 #    elif defined(STM32F303xC) || defined(STM32F401xC)
 #        define FEE_MCU_FLASH_SIZE 256 // Size in Kb
-#    elif defined(STM32F103xE) || defined(STM32F401xE) || defined(STM32F411xE)
+#    elif defined(STM32F103xE) || defined(STM32F303xE) || defined(STM32F401xE) || defined(STM32F411xE)
 #        define FEE_MCU_FLASH_SIZE 512 // Size in Kb
```

Comment:
This is not 100% correct, because the defines that are used here are not guaranteed to match the flash size exactly — if you want to compile a firmware for STM32F303xD (384K flash), you would need to define `STM32F303xE` too. The same situation exists for the F103 series (in that case chips with a larger flash size have also a different flash page size).

However, this is not worse than the existing situation, and `FEE_MCU_FLASH_SIZE` may be redefined at the keyboard level, so this situation may be acceptable for now.

Reply:
So adding more specific defines such as `define FEE_MCU_FLASH_SIZE 384` for chips like the `STM32F303xD` and others would be a more complete solution?

Reply:
A more complete solution would also need to choose a proper linker script, which also needs to explicitly reserve the portion of flash that is used for the EEPROM emulation — currently on most chips except F4x1 the code can overflow into the area used by EEPROM without any errors at the build time. Although adding a size check step (similar to what is used for AVR) could also be an option here (to avoid the need to maintain lots of linker scripts). But all that is probably out of scope of this PR; see links from #8179.